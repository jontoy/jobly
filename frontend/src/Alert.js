import React from 'react';

function Alert({type="danger", msgs=[]}) {
    return (
        <div className={`alert alert-${type}`} role="alert">
            {msgs.map(msg => (
                <p className="small">
                    {msg}
                </p>
            ))}
        </div>
    );
}

export default Alert;