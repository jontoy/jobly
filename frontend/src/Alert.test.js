import React from 'react';
import { render } from '@testing-library/react';
import Alert from './Alert';

it('renders without crashing', () => {
    render(<Alert />);
  });
  
it('matches snapshot with danger messages', () => {
      const {asFragment} = render(<Alert type="danger" msgs={['oh no!', 'it broke!']}/>);
      expect(asFragment()).toMatchSnapshot();
  });
it('matches snapshot with warning messages', () => {
    const {asFragment} = render(<Alert type="warning" msgs={['caution!']}/>);
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot with success messages', () => {
    const {asFragment} = render(<Alert type="success" msgs={[]}/>);
    expect(asFragment()).toMatchSnapshot();
});