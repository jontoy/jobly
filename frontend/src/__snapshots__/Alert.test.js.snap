// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with danger messages 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="small"
    >
      oh no!
    </p>
    <p
      class="small"
    >
      it broke!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with success messages 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  />
</DocumentFragment>
`;

exports[`matches snapshot with warning messages 1`] = `
<DocumentFragment>
  <div
    class="alert alert-warning"
    role="alert"
  >
    <p
      class="small"
    >
      caution!
    </p>
  </div>
</DocumentFragment>
`;
